SwiftWisdom/Core/Timeout/RepeatedTask.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  RepeatedTask.swift
+//  SwiftWisdom
+//
+//  Created by Paul Rolfe on 5/12/17.
+//  Copyright © 2017 Intrepid. All rights reserved.
+//
+
+import Foundation
+
+public class RepeatedTask {
+    let operation: Block
+    let timeBetweenOps: TimeInterval
+    /// nil repeatCount means infinite
+    let repeatCount: Int?
+    let queue: DispatchQueue
+
+    private var currentCount: Int = 0
+    private var isComplete = false
+    private var isActive = false
+
+    public init(repeatCount: Int? = nil, timeBetweenOps: TimeInterval = 0, on queue: DispatchQueue = DispatchQueue.main, operation: @escaping Block) {
+        self.operation = operation
+        self.timeBetweenOps = timeBetweenOps
+        self.repeatCount = repeatCount
+        self.queue = queue
+        start()
+    }
+
+    public func start() {
+        guard !isActive else { return }
+        isActive = true
+        queue.async { [weak self] in
+            self?.executeOp()
+        }
+    }
+
+    public func stop() {
+        guard isActive else { return }
+        isActive = false
+    }
+
+    private func executeOp() {
+        if let count = repeatCount, currentCount >= count {
+            isComplete = true
+            return
```

Comment:
I've been thinking bout that myself.  Is there possibly some value in leaving `isActive` as `true` if the object owner calls `stop` "before" the repeat count has completed (I'm quoting "before," because threading)? That information is opaque to the object owner since `isActive` and `isComplete` are both `private`.  But *should* they be private?  Or maybe they should just be read-only from the outside?